packages/frontend/src/js/app/components/ui/Autocomplete.vue

```diff
@@ -269,24 +281,51 @@ export default {
 
             this.$emit("submit", newValue);
             this.$emit("input", newValue);
+            this.$emit("blur", {
+                value: this.value,
+                search: this.searchInput
+            });
             this.$refs.provider.syncValue(newValue);
             this.$refs.provider.validate();
             this.$refs.searchInput.blur();
+
+            clearTimeout(this.blurTimeout);
+            this.blurTimeout = null;
         },
         handleFocus() {
+            clearTimeout(this.blurTimeout);
+            this.blurTimeout = null;
+
             this.focused = true;
+            this.originalSearchInput = this.searchInput;
         },
 
         handleBlur() {
-            // If user has deleted his input, delete the selected value
-            if (!this.value) {
-                this.onItemSelect(null);
-            } else {
-                // If user has changed his last input, restore to last value
-                this.searchInput = this.getResultValue(this.value);
+            if (this.blurTimeout !== null) {
+                clearTimeout(this.blurTimeout);
             }
-            this.focused = false;
-            this.$emit("blur", { value: this.value, search: this.searchInput });
+
+            this.blurTimeout = setTimeout(() => {
+                // If user has deleted his input, delete the selected value
+                if (!this.value) {
+                    this.onItemSelect(null);
+                } else if (
+                    this.allowFreeInput === true &&
```

Comment:
else if (this.allowFreeInput && ...)
histoire d'être un tatillon, mais c'est un peu redondant d'écrire (someBool === true)